cpp/libcudf_kafka/tests/kafka_consumer_tests.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <map>
+#include <memory>
+#include <string>
+#include "cudf_kafka/kafka_consumer.hpp"
+
+#include <cudf/io/datasource.hpp>
+#include <cudf/io/functions.hpp>
+
+namespace kafka = cudf::io::external::kafka;
+
```

Comment:
This would require introducing Kafka into the CI environment. Would you like for me to tackle that in this PR or a subsequent PR? Keith had mentioned leaving it out until later.

Reply:
I'm a strong -1 to introducing this into the current testing environment. We'll need to work with devops as to how to handle this.

Reply:
cuIO is already under-tested and we end up with a lot of bug reports. If we want to support a feature, we need to test it.

Reply:
Agreed, but testing this code requires spinning up a kafka cluster which is non-trivial and we likely don't want to do in the current CI environment. Once we have pipelined builds I could foresee this testing being a stage in the pipeline **after** libcudf which has a separate container and whatnot which allows isolating the kafka cluster a bit more.